src/nvim/charset.c

```diff
@@ -1394,6 +1394,14 @@ long getdigits_long(char_u **pp, bool strict, long def)
   return (long)number;
 }
 
+/// Gets a int32_t number from a string.
+///
+/// @see getdigits
+int32_t getdigits_int32(char **pp, bool strict, long def)
+{
+  return (int32_t)getdigits((char_u **)pp, strict, def);
```

Comment:
we would need the similar treatment here as the other `getdigits_xx`. something like..

```
  intmax_t number = getdigits(pp, strict, def);
#if SIZEOF_INTMAX_T > sizeof(int32_t)
  if (strict) {
    assert(number >= INT32_MIN && number <= INT32_MAX);
  } else if (!(number >= INT32_MIN && number <= INT32_MAX)) {
    return def;
  }
#endif
  return (int32_t)number;
```